google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/AbstractStatementParser.java

```diff
@@ -696,4 +697,169 @@ static int countOccurrencesOf(char c, String string) {
   public boolean checkReturningClause(String sql) {
     return checkReturningClauseInternal(sql);
   }
+
+  /**
+   * Returns true for characters that can be used as the first character in unquoted identifiers.
+   */
+  boolean isValidIdentifierFirstChar(char c) {
```

Comment:
All the methods that are added here, are only copy-pasted from `PostgreSQLStatementParser`. We do this as step 1 for a larger refactoring, so the next steps are easier to follow and review.